lib/importproject.cpp

```diff
@@ -433,6 +433,10 @@ bool ImportProject::importCompileCommands(std::istream &istr)
             return false;
         }
 
+        //LINKER entries contains a "files" array instead of a single file entry. Skip those without a file.
+        if(!(obj["file"].is<std::string>()))
```

Comment:
The check should probably check for the existence of the field and not if the field is a string.

Reply:
is<type>() Checks if there field is there and is of the expected type. (As I understand it.)

Reply:
obj is a map. so as far as I see `obj["file"]` will create the "file" field if it does not exist.

    typedef std::map<std::string, value> object;

Reply:
Good point. So I guess I need to check the existing code for misuse.

I have not looked at the library behavior but there's a difference between a missing field, a JSON `null` and an empty string (`""`). The first is the one that should be skipped in this case. But I still want to see some documentation of the format IAR outputs first since it does not seen to be a regular compilation database.

Reply:
I think it would be best to check if "file" exists first. I don't know what the default field is.. and maybe that will change in a future version.